pandas/tests/tslibs/test_timedeltas.py

```diff
@@ -37,3 +37,23 @@ def test_huge_nanoseconds_overflow():
     # GH 32402
     assert delta_to_nanoseconds(Timedelta(1e10)) == 1e10
     assert delta_to_nanoseconds(Timedelta(nanoseconds=1e10)) == 1e10
+
+
+# GH40946
+@pytest.mark.parametrize(
+    "obj, expected",
+    [
+        (Timedelta("1us"), 1e-6),
+        (Timedelta("500ns"), 5e-7),
+        (Timedelta(nanoseconds=500), 5e-7),
+        (Timedelta(seconds=1, nanoseconds=500), 1 + 5e-7),
+        # require GH45108
```

Comment:
don't add comment code instead just add this case as a np.timedelta64  constructed from ns ints which preserves this (or add as a followup)

Reply:
Fixed